source/compiler-core/slang-source-loc.cpp

```diff
@@ -573,15 +573,38 @@ int SourceFile::calcColumnIndex(int lineIndex, int offset, int tabSize)
 
 void SourceFile::setContents(ISlangBlob* blob)
 {
-    const UInt contentSize = blob->getBufferSize();
+    const UInt rawContentSize = blob->getBufferSize();
 
-    SLANG_ASSERT(contentSize == m_contentSize);
+    SLANG_ASSERT(rawContentSize == m_contentSize);
 
-    char const* contentBegin = (char const*)blob->getBufferPointer();
-    char const* contentEnd = contentBegin + contentSize;
+    Byte* rawContentBegin = (Byte*)blob->getBufferPointer();
 
-    m_contentBlob = blob;
-    m_content = UnownedStringSlice(contentBegin, contentEnd);
+    // Query the encoding type and discard the Unicode Byte-Order-Marker before decoding
+    size_t offset;
+    auto type = CharEncoding::determineEncoding(
+        rawContentBegin,
+        rawContentSize,
+        offset);
+    SLANG_ASSERT(rawContentSize >= offset);
+
+    List<char> decodedBuffer;
+    CharEncoding::getEncoding(type)->decode(
+        rawContentBegin + offset,
+        rawContentSize - offset,
```

Comment:
This line is giving me a compiler warning,
```
D:\sbf\git\slang\cheneym2\source\compiler-core\slang-source-loc.cpp(594,22): warning C4267: 'argument': conversion from 'size_t' to 'int', possible loss of data [D:\sbf\git\slang\cheneym2\build\compiler-core.vcxproj]
```
Please apply an appropriate type casting to avoid the warning.
```
int(rawContentSize - offset),
```

Reply:
Ack